doc/query.md

```diff
@@ -39,27 +39,60 @@ Secondly, these documents are in the datastore:
 ```java
 { "name": "mike", 
   "age": 12, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Creating Indexes
 
 In order to query documents, indexes need to be created over
-the fields to be queried against.
+the fields to be queried against.  Currently we support two types of indexes.  The first, a JSON index, is used by query clauses containing comparison operators like `$eq`, `$lt`, `$gt`, etc.  Query clauses containing these operators rely on standard SQL syntax along with JSON indexes to provide query results.  The second, a TEXT index, uses SQLite's full text search (FTS) engine.  A query clause containing a `$text` operator with a `$search` operator uses [SQLite FTS SQL syntax] [ftsHome] along with a TEXT index to provide query results.
 
-Use `ensureIndexed(List<Object> fieldNames, String indexName)` to create indexes. These indexes are persistent across application restarts as they are saved to disk. They are kept up to date documents change; there's no need to call `ensureIndexed(List<Object> fieldNames, String indexName)` each time your applications starts, though there is no harm in doing so.
+Using the datastore documents example above, a query like `{ "name" : { "$eq" : "mike" } }` would benefit from a JSON index on the `name` field.  Whereas a query like `{ "$text" : { "$search" : "doctor books" } }` would _require_ a TEXT index on the `comment` field but could include as many fields as desired.  In other words, A TEXT index can be used to search for terms, phrases, prefixes with wildcards, etc.  Query syntax details are covered later in this document.
```

Comment:
Simplify this, I think, and start with by declaring the restriction.

> Basic querying of fields benefits but does _not require_ a JSON index. For example, `{ "name" : { "$eq" : "mike" } }` would benefit from a JSON index on the `name` field but would succeed even if there isn't an index on `name`. Text queries, by contrast, _require_ an index. Therefore `{ "$text" : { "$search" : "doctor books" } }` would need a TEXT index on the `comment` field (based on the content of the above documents). A TEXT index can be used to search for terms, phrases and prefixes using wildcards. Querying capabilities and syntax are covered later in this documents.

Don't use `etc.` because it's impossible for a user to infer what you mean -- otherwise they probably wouldn't need to read the docs.

I'm not sure what "could include as many fields as desired" means?